src/options/App.tsx

```diff
@@ -93,57 +92,64 @@ function OptionsPage(props: { theme: Theme; onThemeChange: (theme: Theme) => voi
-        />
-
-        {promptOverrides.map((override, index) => {
-          return (
-            <div key={override.site} className="my-3">
-              <PromptCard
-                header={override.site}
-                prompt={override.prompt}
-                onSave={(newPrompt) => {
-                  const newOverride: SitePrompt = {
-                    site: override.site,
-                    prompt: newPrompt,
-                  }
-                  const newOverrides = promptOverrides.filter((o) => o.site !== override.site)
-                  newOverrides.splice(index, 0, newOverride)
-                  setPromptOverrides(newOverrides)
-                  return updateUserConfig({ promptOverrides: newOverrides })
-                }}
-                onDismiss={() => {
-                  const newOverrides = promptOverrides.filter((_, i) => i !== index)
-                  setPromptOverrides(newOverrides)
-                  return updateUserConfig({ promptOverrides: newOverrides })
-                }}
-              />
-            </div>
-          )
-        })}
+        {supportedURLs &&
+          supportedURLs.map((URL, index) => {
+            return (
+              <div key={URL} className="my-3">
+                <SiteCard
+                  prompt={siteConfigs[URL].prompt}
+                  URL={URL}
+                  bodyTag={siteConfigs[URL].bodyTag}
+                  displayTag={siteConfigs[URL].displayTag}
+                  onSave={(URL, prompt, bodyTag, displayTag) => {
+                    supportedURLs[index] = URL
+                    siteConfigs[URL] = {
+                      prompt: prompt,
+                      bodyTag: bodyTag,
+                      displayTag: displayTag,
+                      URL: URL,
+                    }
+                    return updateUserConfig({
+                      supportedURLs: supportedURLs,
+                      siteConfigs: siteConfigs,
+                    })
+                  }}
+                  onDismiss={() => {
+                    supportedURLs.splice(index, 1)
+                    delete siteConfigs[URL]
+                    return updateUserConfig({
+                      supportedURLs,
+                      siteConfigs,
+                    })
+                  }}
+                />
+              </div>
+            )
+          })}
 
         <Button mt={1} type="secondary" width={'100%'} onClick={() => setModalVisible(true)}>
           <Plus size={16} className="mx-2" />
-          Add Prompt
+          Add Site Config
         </Button>
 
-        <AddNewPromptModal
+        <AddNewSiteConfigModal
           visible={modalVisible}
           onClose={closeModalHandler}
-          onSave={({ site, prompt }) => {
-            const newOverride: SitePrompt = {
-              site,
-              prompt,
+          supportedURLs={supportedURLs}
+          onSave={(update) => {
+            supportedURLs[supportedURLs.length] = update.site
```

Comment:
It's always better to keep objects immutable. Please create a new object from old one and use it instead.